# Importing Modules and Checking Directory 

In [20]:
import os
import pandas as pd
import datetime
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
%matplotlib inline

In [21]:
pd.__version__

'1.1.5'

In [22]:
os.getcwd()


'/Users/Livi/Projects/Black_mental_health/notebooks/Data_wrangling'

# Uploading Data 

In [23]:
census = pd.read_csv('../../data/raw/2017acs.csv')

In [24]:
census.head()

geo_id  nonfamily_households  family_households  \
0   55111                    13                  0   
1   61112                    93                  7   
2   33856                   189                 84   
3   32079                   151                 69   
4   85633                    12                  4   

   median_year_structure_built  rent_burden_not_computed  \
0                       2016.0                         4   
1                       2001.0                         0   
2                       1966.0                         0   
3                       1970.0                        74   
4                       1939.0                         2   

   rent_over_50_percent  rent_40_to_50_percent  rent_35_to_40_percent  \
0                     5                      0                      0   
1                    57                     15                      0   
2                     0                     59                     18   
3                    49                      3                      8   
4                     0                      0                      3   

   rent_30_to_35_percent  rent_25_to_30_percent  ...  \
0                      4                      0  ...   
1                      0                      0  ...   
2                      7                     62  ...   
3                     17                     20  ...   
4                      0                      0  ...   

   speak_only_english_at_home  speak_spanish_at_home  \
0                         NaN                    NaN   
1                         NaN                    NaN   
2                         NaN                    NaN   
3                         NaN                    NaN   
4                         NaN                    NaN   

   speak_spanish_at_home_low_english  pop_15_and_over  pop_never_married  \
0                                NaN              NaN                NaN   
1                                NaN              NaN                NaN   
2                                NaN              NaN                NaN   
3                                NaN              NaN                NaN   
4                                NaN              NaN                NaN   

   pop_now_married  pop_separated  pop_widowed  pop_divorced   do_date  
0              NaN            NaN          NaN           NaN  20132017  
1              NaN            NaN          NaN           NaN  20132017  
2              NaN            NaN          NaN           NaN  20132017  
3              NaN            NaN          NaN           NaN  20132017  
4              NaN            NaN          NaN           NaN  20132017  

[5 rows x 252 columns]

checkpoint2

In [25]:
checkpoint2 = census.copy()

In [26]:
list(census.columns[census.columns.str.contains('pop')])

['total_pop',
 'male_pop',
 'female_pop',
 'white_pop',
 'black_pop',
 'asian_pop',
 'hispanic_pop',
 'amerindian_pop',
 'other_race_pop',
 'two_or_more_races_pop',
 'not_hispanic_pop',
 'employed_pop',
 'unemployed_pop',
 'pop_16_over',
 'pop_in_labor_force',
 'not_us_citizen_pop',
 'pop_25_64',
 'pop_determined_poverty_status',
 'population_1_year_and_over',
 'population_3_years_over',
 'pop_25_years_over',
 'pop_5_years_over',
 'pop_15_and_over',
 'pop_never_married',
 'pop_now_married',
 'pop_separated',
 'pop_widowed',
 'pop_divorced']

In [27]:
list(census.columns[census.columns.str.contains('rent|income', regex=True)])

['rent_burden_not_computed',
 'rent_over_50_percent',
 'rent_40_to_50_percent',
 'rent_35_to_40_percent',
 'rent_30_to_35_percent',
 'rent_25_to_30_percent',
 'rent_20_to_25_percent',
 'rent_15_to_20_percent',
 'rent_10_to_15_percent',
 'rent_under_10_percent',
 'median_income',
 'income_per_capita',
 'vacant_housing_units_for_rent',
 'median_rent',
 'percent_income_spent_on_rent',
 'two_parent_families_with_young_children',
 'two_parents_in_labor_force_families_with_young_children',
 'two_parents_father_in_labor_force_families_with_young_children',
 'two_parents_mother_in_labor_force_families_with_young_children',
 'two_parents_not_in_labor_force_families_with_young_children',
 'one_parent_families_with_young_children',
 'father_one_parent_families_with_young_children',
 'father_in_labor_force_one_parent_families_with_young_children',
 'income_less_10000',
 'income_10000_14999',
 'income_15000_19999',
 'income_20000_24999',
 'income_25000_29999',
 'income_30000_34999',
 'income_35000_

How many zips are missing aggregate information on income or rent that could help determine the cost of living per zip?

In [28]:
cols = ['median_income', 'income_per_capita',
 'median_rent',]
census[cols].isnull().sum()

median_income        2164
income_per_capita     642
median_rent          6102
dtype: int64

642 is a low number of zip codes (out of 30,000) but it would skew my data of providers to have to remove all providers in these 642 zips. A future version of this project could impute income_per_capita in these zips to include income data in my models

In [29]:
cols =['geo_id','total_pop',
 'white_pop',
 'black_pop',
 'asian_pop',
 'hispanic_pop',
 'amerindian_pop',
 'other_race_pop',
 'two_or_more_races_pop','do_date']

In [30]:
census = census[cols]

In [31]:
info = pd.DataFrame(index=census.columns.values,
                    data={'data type': census.dtypes, \
                          'non-null values': census.notnull().sum(),
                          'unique values': census.nunique(), \
                          'null values': census.isnull().sum()})
info.sort_values('non-null values').head(30)

data type  non-null values  unique values  null values
geo_id                    int64            33120          33120            0
total_pop                 int64            33120          15566            0
white_pop                 int64            33120          13181            0
black_pop                 int64            33120           5320            0
asian_pop                 int64            33120           3489            0
hispanic_pop              int64            33120           6095            0
amerindian_pop            int64            33120           1063            0
other_race_pop            int64            33120            536            0
two_or_more_races_pop     int64            33120           2050            0
do_date                   int64            33120              1            0

# Checking Data Types

All columns in Census besides Geo_id should be 'int' or 'float.' I want to see what is listed as object.

In [32]:
census.select_dtypes('object').head(1)

Empty DataFrame
Columns: []
Index: [0]

In [33]:
census.do_date

0        20132017
1        20132017
2        20132017
3        20132017
4        20132017
           ...   
33115    20132017
33116    20132017
33117    20132017
33118    20132017
33119    20132017
Name: do_date, Length: 33120, dtype: int64

There's a column called do_date that includes the years used to make the estimate Census measures. I'm going to keep that last four digits which represent the year being estimated.

In [34]:
census.do_date = census.do_date.apply(lambda x: str(x)[-4:])
census.do_date = pd.to_datetime(census.do_date, format='%Y')
census.do_date

0       2017-01-01
1       2017-01-01
2       2017-01-01
3       2017-01-01
4       2017-01-01
           ...    
33115   2017-01-01
33116   2017-01-01
33117   2017-01-01
33118   2017-01-01
33119   2017-01-01
Name: do_date, Length: 33120, dtype: datetime64[ns]

In [35]:
census.to_csv('../../data/processed/2017pops.csv')